### ISSUE TITLE
close() and quit() not working with selenium 3.7.1

### ISSUE THREAD

**Issue by hornschorsch** [teodesian/Selenium-Remote-Driver#346]
Hello,

another one with selenium 3.7.1. Close and quit are no longer working.

```
REQ: DELETE, http://localhost:4444/wd/hub/session/853154c0-a6ed-410a-8983-963651b0727c/window, {}
RES: {"state":"unknown error","sessionId":null,"hCode":1060642583,"value":{"additionalInformation":"\nDriver info: driver.version: unknown","localizedMessage":"Failed to interpret value as array\nBuild info: version: '3.5.3', revision: 'a88d25fe6b', time: '2017-08-29T12:54:15.039Z'\nSystem info: host: 'mozart', ip: '192.168.100.5', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.17', java.version: '1.8.0_111'\nDriver info: driver.version: unknown","systemInformation":"System info: host: 'mozart', ip: '192.168.100.5', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.17', java.version: '1.8.0_111'","supportUrl":null,"cause":null,"stackTrace":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"suppressed":[],"message":"Failed to interpret value as array\nBuild info: version: '3.5.3', revision: 'a88d25fe6b', time: '2017-08-29T12:54:15.039Z'\nSystem info: host: 'mozart', ip: '192.168.100.5', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.17', java.version: '1.8.0_111'\nDriver info: driver.version: unknown","hCode":793292219,"class":"org.openqa.selenium.WebDriverException","buildInformation":null},"class":"org.openqa.selenium.remote.Response","status":13}
```

```
REQ: DELETE, http://localhost:4444/wd/hub/session/103c8b33-cff1-48ee-a981-aba29dc5179e, {}
RES: {"state":"unknown error","sessionId":null,"hCode":1022341447,"value":{"additionalInformation":"\nDriver info: driver.version: unknown","localizedMessage":"quit\nBuild info: version: '3.5.3', revision: 'a88d25fe6b', time: '2017-08-29T12:54:15.039Z'\nSystem info: host: 'mozart', ip: '192.168.100.5', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.17', java.version: '1.8.0_111'\nDriver info: driver.version: unknown","systemInformation":"System info: host: 'mozart', ip: '192.168.100.5', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.17', java.version: '1.8.0_111'","supportUrl":null,"cause":null,"stackTrace":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"suppressed":[],"message":"quit\nBuild info: version: '3.5.3', revision: 'a88d25fe6b', time: '2017-08-29T12:54:15.039Z'\nSystem info: host: 'mozart', ip: '192.168.100.5', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.17', java.version: '1.8.0_111'\nDriver info: driver.version: unknown","hCode":1347126993,"class":"org.openqa.selenium.WebDriverException","buildInformation":null},"class":"org.openqa.selenium.remote.Response","status":13}
```

**Comment by hornschorsch** [comment]
A short update, this only happens with geckodriver 0.16.1 and greater.

with 0.15 and 0.14 it works but still prints:

```
Couldn't retrieve command settings properly
 at /usr/local/share/perl5/Selenium/Remote/Driver.pm line 327.
 at /usr/local/share/perl5/Selenium/Remote/Driver.pm line 327.
```

**Comment by teodesian** [comment]
Unknown Error: Failed to interpret value as array
and 
Unknown Error: quit

This sounds like geckodriver 0.16.1 is buggy.  Maybe there's a workaround.

**Comment by hornschorsch** [comment]
It also fails with geckodriver 0.17 and 0.18. I now updated to 0.19.1 and firefox 55 (which is needed by geckodriver 0.19, i hate this versionnnumber racing...) and now my testscript at least works again, but it still prints an error after the $driver-> close();

```
REQ: DELETE, http://localhost:4444/wd/hub/session/38eb6ca0-6ed6-4a04-a736-889327ae724f, {}
RES: {"state":"success","sessionId":null,"hCode":1227042721,"value":{},"class":"org.openqa.selenium.remote.Response","status":0}

Couldn't retrieve command settings properly
 at /usr/local/share/perl5/Selenium/Remote/Driver.pm line 327.
 at /usr/local/share/perl5/Selenium/Remote/Driver.pm line 327.
```
the selenium server log looks OK for me:

```
13:48:13.175 INFO - Handler thread for session 38eb6ca0-6ed6-4a04-a736-889327ae724f (firefox): Executing DELETE on /session/38eb6ca0-6ed6-4a04-a736-889327ae724f (handler: ServicedSession)
1512391693178   Marionette      INFO    New connections will no longer be accepted
13:48:13.423 INFO - Removing session org.openqa.selenium.remote.server.ServicedSession@2bc0f6d3
```

**Comment by hornschorsch** [comment]
One last coment, the message "Couldn't retrieve command settings properly" was caused by my stupid script, which tried to capture a screenshot after closing the browser...

So everything is fine with geckodriver-0.19.1, selenium-3.7.1 and firefox-55.

**Comment by teodesian** [comment]
Likely dupe of #345, leaving open until done there "just to be sure"

**Comment by teodesian** [comment]
yup, have fixed in the branch for #345
